grails-app/services/org/pih/warehouse/inventory/OutboundStockMovementService.groovy

```diff
@@ -143,6 +150,6 @@ class OutboundStockMovementService {
             }
         }
 
-        return stockMovements
+        return stockMovements.unique()
```

Comment:
this is not good in my opinion. Pagination might break at this point.

Reply:
you are right, this needs a different approach, unfortunately using `.distinctList()` does not work with paginated arguments such as `max` and `offset` so probably I'll need to fetch the `totalCount` separately

Reply:
it's a pain in the neck, but I believe you can follow what I've suggested to @alannadolny once a long time ago, but I don't remember where it was. Probably something connected to products? Where we've built two createCriterias but one of them was just returning a `totalCount`, and one of them was returning the List.
@alannadolny do you remember where it was, so that @drodzewicz could follow the approach?

Reply:
Do you mean this pr? https://github.com/openboxes/openboxes/pull/3859